apps/studio/components/interfaces/Organization/TeamSettings/LeaveTeamButton.tsx

```diff
@@ -0,0 +1,121 @@
+import { useRouter } from 'next/router'
+import { useState } from 'react'
+import { toast } from 'sonner'
+
+import { LOCAL_STORAGE_KEYS, useParams } from 'common'
+import { ButtonTooltip } from 'components/ui/ButtonTooltip'
+import { useOrganizationRolesV2Query } from 'data/organization-members/organization-roles-query'
+import { useOrganizationMemberDeleteMutation } from 'data/organizations/organization-member-delete-mutation'
+import { useOrganizationMembersQuery } from 'data/organizations/organization-members-query'
+import { useOrganizationsQuery } from 'data/organizations/organizations-query'
+import { useIsFeatureEnabled } from 'hooks/misc/useIsFeatureEnabled'
+import { useLocalStorageQuery } from 'hooks/misc/useLocalStorage'
+import { useSelectedOrganization } from 'hooks/misc/useSelectedOrganization'
+import { useProfile } from 'lib/profile'
+import ConfirmationModal from 'ui-patterns/Dialogs/ConfirmationModal'
+import { hasMultipleOwners } from './TeamSettings.utils'
+
+export const LeaveTeamButton = () => {
+  const router = useRouter()
+  const { slug } = useParams()
+  const { profile } = useProfile()
+  const selectedOrganization = useSelectedOrganization()
+
+  // if organizationMembersDeletionEnabled is false, you also can't delete yourself
+  const { organizationMembersDelete: organizationMembersDeletionEnabled } = useIsFeatureEnabled([
+    'organization_members:delete',
+  ])
+
+  const [isLeaving, setIsLeaving] = useState(false)
```

Comment:
Shouldn't this `isLoading` on the mutation?

Reply:
ahh yes we're using a useState here as i'd want the button to still be disabled while we're re-refetching the organizations + re-routing the user

if we were to use `isLoading` from the mutation, the button will undo the disabled state too early